### PR TITLE
feat: add README and tweak Makefile to update workspace cargo file

### PR DESCRIPTION
**Pull request by quake**
1. added cargo-generate min version requirement ( >=0.16.0)
2. added README template
3. auto update workspace cargo file when generate sub crate and contract

### CODE REVIEW

File: post-script.rhai [hunk 1]
```diff
@@ -0,0 +1 @@
+system::command("touch", ["../test.txt"])
```
Comment (xxuejie) [inline post-script.rhai:1]: What is this used for?

File: atomics-contract/cargo-generate.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[template]
+cargo_generate_version = ">=0.16.0"
```
Comment (xxuejie) [inline atomics-contract/cargo-generate.toml:2]: Any chance we can explain why 0.16.0 version is picked here?

File: post-script.rhai (hunk 1 above)
Comment (quake) [reply post-script.rhai]: sorry, checked in by mistake.

File: atomics-contract/cargo-generate.toml (hunk 2 above)
Comment (quake) [reply atomics-contract/cargo-generate.toml:2]: related to sub-template feature:
https://github.com/cargo-generate/cargo-generate/blob/main/CHANGELOG.md#-features-6

**Review by xxuejie — APPROVED** [review]

**Pull request merged**